src/components/Typography/BodyX/BodyX.tsx

```diff
@@ -46,18 +47,28 @@ export const BodyX = ({
   children,
   copyable = defaults.copyable,
   ellipsis = defaults.ellipsis,
+  hierarchy = defaults.hierarchy,
   isBold = defaults.isBold,
   size,
 }: BodyXProps & BodyXSize): ReactElement => {
-  const bodyClassName = useMemo(() => classnames([
+  const sizeClassName = useMemo(() => classnames([
     size === S && bodyS,
     size === M && bodyM,
     size === L && bodyL,
   ]), [size])
+  const hierarchyClassName = useMemo(() => classnames([
+    hierarchy === Hierarchy.Bold && bold,
+    hierarchy === Hierarchy.Main && main,
```

Comment:
I'd suggest not injecting any class if main is set

Since:

- applying this default changes tons of snapshots (we see some change here but it will also affect all of the snapshots for the projects that are using this component)
- the main class is currently empty